ldelf/ta_elf_rel.c

```diff
@@ -414,6 +433,14 @@ static void e64_get_sym_name(const Elf64_Sym *sym_tab, size_t num_syms,
 	if (name_idx >= str_tab_size)
 		err(TEE_ERROR_BAD_FORMAT, "Name index out of range");
 	*name = str_tab + name_idx;
+
+	if (!weak_undef)
```

Comment:
This is only non-NULL. Wouldn't it make more sense to always assume it's non-NULL?